### ISSUE TITLE
Calling WinVerifyTrust concurrently Results in Client Crashing

### ISSUE THREAD

**Issue by chaoticmachinery** [Velocidex/velociraptor#2574]
I am seeing stuck Hunts on some of the clients. The hunt says it is running and I have the limits set to unlimited. There is nothing in the logs regarding the issue. To simulate the issue I ran the same hunt three times on a single client scheduled one right after another. The first two completed, but third is just been sitting there for the last 30 min. I did enable trace for every 10 second. All of the trace files are null. The test client system doesn't have a load on it (CPU is around 5%). The first two hunts completed without issue. Plus I am able to use VFS to browse the file system even though the hunt job is stuck.  

To further investigate the issue, stopped VR as a service and ran it from the command line as:
"C:\Program Files\Velociraptor\Velociraptor.exe"  --config "C:\Program Files\Velociraptor\client.config.yaml" client -v > e:\raptor2.txt 2>&1

This time I repeated the test and multiples of the same hunt. I couldn't think of another way to replicate what I am seeing. The client did crashed.  Please let me know if you need any more information.

Below is the output of the "-v" option and the error messages. 
 
[raptor2.txt](https://github.com/Velocidex/velociraptor/files/11044184/raptor2.txt)

**Comment by chaoticmachinery** [comment]
Client machine was: Microsoft Windows 10 Enterprise10.0.19045 Build 19045  64-bit

To reproduce:
1) Schedule a hunt with the AmcacheGPS.yaml
2) Right after scheduling repeat step 1 four to eight times. Do not wait for the first job to complete.

Again, I was trying to figure out why sometimes this hunt (and similar ones) wouldn't do anything for some clients where as the default one amcache hunt would work. Doing scheduling of multiple same hunts replicated the behavior I was seeing in the field.  

[AmcacheGPS.yaml.txt](https://github.com/Velocidex/velociraptor/files/11050722/AmcacheGPS.yaml.txt)

**Comment by chaoticmachinery** [comment]
[raptor4.txt](https://github.com/Velocidex/velociraptor/files/11051288/raptor4.txt)
Got another crash related to this using the binary provided by Mr. Cohen per discussion on discord.

**Comment by chaoticmachinery** [comment]
[raptor7.txt](https://github.com/Velocidex/velociraptor/files/11053996/raptor7.txt)
Crashed. Attached is the cmd output.

**Comment by chaoticmachinery** [comment]
[Trace1679592823.zip](https://github.com/Velocidex/velociraptor/files/11054549/Trace1679592823.zip)
Adding trace file

**Comment by scudette** [comment]
Submitted #2582 to allow disabling this API call completely.

You will need to add the following parameter to the artifact definition
```yaml
  - name: DISABLE_DANGEROUS_API_CALLS
    type: bool
    default: Y
```

**Comment by chaoticmachinery** [comment]
[Trace1679751774.zip](https://github.com/Velocidex/velociraptor/files/11069152/Trace1679751774.zip)
[raptor20.txt](https://github.com/Velocidex/velociraptor/files/11069153/raptor20.txt)
03/25/23 tests results

**Comment by scudette** [comment]
Should be fixed in 0.6.8-2 - we still get the occasional crash but now it is possible to disable this API selectively. This is probably the best we can do if the underlying API is buggy and not thread safe
